### ISSUE TITLE
Table meta information

### ISSUE THREAD

**Issue by mw25** [biolab/orange3#6333]
**What's your use case?**

In general: I would like to attach meta information to a table, which can be used in the following widgets.

I have seen that Table has `attributes`. But I am not sure what you have provided that for and if I can use it for my purposes.

Also, the table loses its attributes in some widgets that create completely different output data from the input data (like owgroupby or owpivot). But for me it is quite essential that the table keeps the information.

My case in concrete terms:
In my importer, I would like to append the file name to the table so that the file name can be displayed in a visualisation widget, e.g. as a title. Even if owgroupby changes the data in between, the file name should still be kept.

**What's your proposed solution?**

No real proposal, but at the moment I'm using the following prototype approach:

I have monkeypatched orangewidget.utils.signals.Output.send so that the table attribute "meta_information" (dict that also includes the file name) is inserted into the output data if it is present at the input.
This way I make sure that the information is passed on, regardless of what the widget does.

**Are there any alternative solutions?**

I don't know. Ensure that `attributes` are passed on for all widgets?

To summarise:
- Does such a behaviour already exist in Orange?
- Are you interested in including it in Orange?
- Do you have a better solution?

**Comment by markotoplak** [comment]
As you noticed, Orange already copies `.name` and `.attributes` whenever a table is transformed (trough `.transform` or `Table.from_table`, so most widgets do keep them. But some widgets do create tables in other ways; there this info could get lost.

We discussed this today and we found cases where copying `.attributes` could be misleading, so we should look at this widget-per-widget basis. Which widgets you had problems with?

You already mentioned Pivot and Group By. Personally, I do not see any reasons why these two should not pass `.attributes` and `.name` forward, so that is likely a bug.

**Comment by mw25** [comment]
I have only had problems with Group By so far.

As you say, it would have to be decided on a case-by-case basis.
I have looked for widgets that do not pass on the attributes. And I have prepared the following list for the widget-per-widget comparison:

(No guarantee of completeness. I looked for widgets that use Table(), Table.from_domain(), Table.from_numpy(), Table.from_list() for their output. I would not have found Group By Widget with it, for example).

- PCA
	- Data
	- Components
	- (Transformed Data keeps the attributes)
- Distributions
	- Histogram Data
- Venn Diagram 
	- both outputs
- Linear Regression
	- Coefficients
- Stochastic Gradient Descent
	- Coefficients
- Logistic Regression
	- Coefficients
- Paint Data
- Melt
- Create Instance
	- Without Option "Append this instance to input Data"
- Pivot Table
	- Pivot Table
	- Grouped Data
- Feature Statistics
	- Statistics
- Correlations
	- Correlations
- Rank
	- Scores
- Manifold Learning
- Group by



I also noticed that Merge Data only takes the attributes from Data Input, while Concatenate adds the attributes from both inputs.

For my particular use case, it would be handy if when merging tables, the attribute-dict is not only updated, but values of a key that occurs multiple times are appended if possible (if the value is a list, for example). In my case, this would make it clear that the data does not only come from one file.
But I understand if this is too specific.

Can I help with the implementation?
